node/service/src/rpc/middleware.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! RPC Middleware
+
+use jsonrpc_core::{
+	futures::{future::Either, Future},
+	middleware::NoopCallFuture,
+	Id,
+};
+use jsonrpc_core::{FutureResponse, Middleware, Output, Request, Response};
+use jsonrpc_pubsub::PubSubMetadata;
+use std::sync::atomic::{self, AtomicUsize};
+use std::time::Instant;
+
+type Meta = sc_rpc::Metadata;
+
+#[derive(Default)]
+pub struct MoonbeamMiddleware(AtomicUsize);
+impl Middleware<Meta> for MoonbeamMiddleware {
+	type Future = FutureResponse;
+	type CallFuture = NoopCallFuture;
+
+	fn on_request<F, X>(&self, request: Request, meta: Meta, next: F) -> Either<Self::Future, X>
+	where
+		F: Fn(Request, Meta) -> X + Send + Sync,
+		X: Future<Item = Option<Response>, Error = ()> + Send + 'static,
+	{
+		let start = Instant::now();
+		let request_number = self.0.fetch_add(1, atomic::Ordering::SeqCst);
+		log::debug!(
+			"Processing request {}: {:?}, {:?}",
```

Comment:
Is there the IP in the request that we could include ?

Reply:
Not in the request, we could pass it in the MetaData, jsonrpc let us free to define what we want, but unfortunately substrate imposes that the metadata is of type sc_rpc::Metadata, ec this type does not expose the IP :/